### PR TITLE
MySQL/MariaDB: test with wkt data series

### PR DESCRIPTION
**Pull request by miurahr**
## What does this PR do?

AutoTest writing and reading back geometries by get_wkt_data_series() for MySQL.
It succeeded with MySQL 8 but  fails with MariaDB.

## What are related issues/pull requests?

#2130 

## Tasklist

 - [x] Add test case(s)
 - [x] Investigate root cause
 - [x] Report to upstream
 - [ ] fix a problem
 - [x] Review
 - [x] Adjust for comments
 - [ ] All CI builds and checks have passed

## Environment

MariaDB

### CODE REVIEW

**Comment by miurahr** [comment]
Following error happened in MariaDB test.
```
----------------------------- Captured stdout call -----------------------------
ogr_mysql_28: total 84 geometries
----------------------------- Captured stderr call -----------------------------
ERROR 1: MySQL error message:Column 'SHAPE' cannot be null Description: INSERT INTO `ogr_mysql_28_15` (`SHAPE` ) VALUES (ST_GeomFromText('GEOMETRYCOLLECTION (POINT (4 6),GEOMETRYCOLLECTION (POINT (4 6),LINESTRING (4 6,7 10)))',2) )
```

**Comment by miurahr** [comment]
Report to MariaDB dev https://jira.mariadb.org/browse/MDEV-21401

**Comment by miurahr** [comment]
MariaDB ticket has been confirmed and assigned in 12, Jan, but no progress in end of Jan.

**Comment by miurahr** [comment]
MariaDB ticket has been confirmed and assigned in 12, Jan, but no progress in 21, Feb.

**Comment by miurahr** [comment]
Now I have asked to the upstream tracker about progress because of stale over 50 days after upstream confirmation.

**Comment by rouault** [comment]
@miurahr If you don't want forever for MySQL upstream to make progress on this PR, perhaps mark as xfail / skip the geometries that cause issues with the server ?

File: autotest/ogr/ogr_mysql.py [hunk 1]
```diff
@@ -914,10 +914,67 @@ def test_ogr_mysql_longlat():
...
+# Test writing and reading back geometries
+
+@pytest.mark.xfail(reason="MariaDB has a known issue MDEV-21401")
+def test_ogr_mysql_28():
+
+    if gdaltest.mysql_ds is None:
+        pytest.skip()
+
+    wkts = ogrtest.get_wkt_data_series(with_z=True, with_m=True, with_gc=True, with_circular=True, with_surface=False)
+    print('ogr_mysql_28: total %d geometries' % len(wkts))
```
Comment (rouault) [inline autotest/ogr/ogr_mysql.py]: remove that print()

File: autotest/ogr/ogr_mysql.py [hunk 2]
```diff
@@ -914,10 +914,67 @@ def test_ogr_mysql_longlat():
...
+
+
+@pytest.mark.xfail(reason='MySQL does not support POLYHEDRALSURFACE.')
+def test_ogr_mysql_29():
+
+    if gdaltest.mysql_ds is None:
+        pytest.skip()
+
+    wkts = ogrtest.get_wkt_data_series(with_z=False, with_m=False, with_gc=False, with_circular=False, with_surface=True)
+    print('ogr_mysql_29: total %d geometries' % len(wkts))
```
Comment (rouault) [inline autotest/ogr/ogr_mysql.py]: remove that print()

**Pull request merged**